odl/discr/diff_ops.py

```diff
@@ -698,71 +783,63 @@ def finite_diff(f, axis=0, dx=1.0, method='forward', out=None, **kwargs):
          An N-dimensional array to which the output is written. Has to have
          the same shape as the input array ``f``.
 
-    pad_mode : {'constant', 'symmetric', 'periodic', ``None``}, optional
+    pad_mode : string, optional
```

Comment:
Also in this function, `axis=0` as default makes no sense at all. It should be a mandatory argument.